pkg/network/ebpf/c/protocols/events.h

```diff
@@ -91,6 +91,21 @@
         if (name##_batch_full(batch)) {                                                 \
             batch_state->idx++;                                                         \
         }                                                                               \
+    }                                                                                   \
+                                                                                        \
+    static __always_inline void name##_batch_pop() {                                    \
+        u32 zero = 0;                                                                   \
+        batch_state_t *batch_state =  bpf_map_lookup_elem(&name##_batch_state, &zero);  \
+        if (batch_state == NULL) {                                                      \
+            return;                                                                     \
+        }                                                                               \
+                                                                                        \
+        batch_key_t key = get_batch_key(batch_state->idx);                              \
+        batch_data_t *batch = bpf_map_lookup_elem(&name##_batches, &key);               \
+        if (batch == NULL || batch->len == 0) {                                         \
+            return;                                                                     \
+        }                                                                               \
+        batch->len--;                                                                   \
```

Comment:
not sure I fully understand why is it solving the scenario

Reply:
another suggestion, if our purpose is to avoid sending a batch twice, maybe we can use some unique index to mark traffic
and in the user mode to filter them

Reply:
`batch->len--` is equivalent to removing the last element from the batch so it has the same "semantics" of `doing a pop.
as to why this solves the problem maybe https://github.com/DataDog/datadog-agent/pull/15609#discussion_r1106298206 helps clarifying it?

Reply:
> another suggestion, if our purpose is to avoid sending a batch twice, maybe we can use some unique index to mark traffic and in the user mode to filter them
IMO that sounds much more overkill (and inefficient) to me